homeassistant/components/tplink/number.py

```diff
@@ -0,0 +1,108 @@
+"""Support for TPLink number entities."""
+
+from __future__ import annotations
+
+import logging
+from typing import Final
+
+from kasa import Device, Feature
+
+from homeassistant.components.number import (
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TPLinkConfigEntry
+from .entity import (
+    CoordinatedTPLinkFeatureEntity,
+    TPLinkDataUpdateCoordinator,
+    TPLinkFeatureEntityDescription,
+    async_refresh_after,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TPLinkNumberEntityDescription(
+    NumberEntityDescription, TPLinkFeatureEntityDescription
+):
+    """Base class for a TPLink feature based sensor entity description."""
+
+
+NUMBER_DESCRIPTIONS: Final = (
+    TPLinkNumberEntityDescription(
+        key="smooth_transition_on",
+        mode=NumberMode.BOX,
+    ),
+    TPLinkNumberEntityDescription(
+        key="smooth_transition_off",
+        mode=NumberMode.BOX,
+    ),
+    TPLinkNumberEntityDescription(
+        key="auto_off_minutes",
+        mode=NumberMode.BOX,
+    ),
+    TPLinkNumberEntityDescription(
+        key="temperature_offset",
+        mode=NumberMode.BOX,
+    ),
+)
+
+NUMBER_DESCRIPTIONS_MAP = {desc.key: desc for desc in NUMBER_DESCRIPTIONS}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: TPLinkConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors."""
+    data = config_entry.runtime_data
+    parent_coordinator = data.parent_coordinator
+    children_coordinators = data.children_coordinators
+    device = parent_coordinator.device
+    entities = CoordinatedTPLinkFeatureEntity.entities_for_device_and_its_children(
+        device=device,
+        coordinator=parent_coordinator,
+        feature_type=Feature.Type.Number,
+        entity_class=Number,
+        descriptions=NUMBER_DESCRIPTIONS_MAP,
+        child_coordinators=children_coordinators,
+    )
+
+    async_add_entities(entities)
+
+
+class Number(CoordinatedTPLinkFeatureEntity, NumberEntity):
```

Comment:
```suggestion
class TPLinkNumberEntity(CoordinatedTPLinkFeatureEntity, NumberEntity):
```

Reply:
Done in https://github.com/home-assistant/core/pull/120404